src/main/java/seedu/address/logic/commands/AddFavouriteCommand.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.EditCommand.EditPersonDescriptor;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds the indicated contacts in the address book as favourites
+ */
+public class AddFavouriteCommand extends Command {
+    public static final String COMMAND_WORD = "addfav";
+
+    public static final String MESSAGE_SUCCESS = "The following contacts have been added to favourites: %s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds contacts identified by index number "
+            + "as favourites.\n"
+            + "Parameters: i/ [INDICES] (must be positive integers separated by commas)\n"
+            + "Example: " + COMMAND_WORD + " "
+            + "i/ 1,2,5";
+
+    private final List<Index> indices;
+
+    /**
+     * Creates an AddFavouriteCommand to mark the specified group of {@code Person} as favourites
+     */
+    public AddFavouriteCommand(List<Index> indices) {
+        requireNonNull(indices);
+        this.indices = indices;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> people = model.getFilteredPersonList();
+        List<String> modifiedContacts = new ArrayList<>();
+        for (Index index : this.indices) {
+            if (index.getZeroBased() >= people.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            Person person = people.get(index.getZeroBased());
+            modifiedContacts.add(person.getName().fullName);
```

Comment:
Would it be better if you check for unique indexes?

![image](https://github.com/AY2324S2-CS2103T-T16-3/tp/assets/76832850/9276ed30-f1bd-4b5e-b1ef-e52f8edf626c)

Reply:
Modified the command to take in a set instead of a list to handle duplicate indices